hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/ObjectEndpoint.java

```diff
@@ -1389,4 +1399,11 @@ public MessageDigest getMessageDigestInstance() {
     return E_TAG_PROVIDER.get();
   }
 
+  private boolean isMultipartETag(String eTag) {
+    return eTag != null && eTag.contains("-");
```

Comment:
The condition `eTag != null` here seems redundant. 
How about removing the check here and refactoring the second condition under `@GET`: `L493` and `@HEAD`: `L622` and only then add the response header field?